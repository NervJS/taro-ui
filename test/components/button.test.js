import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtButton from '../../.temp/components/button/index'

describe('AtButton Snap', () => {
  it('render AtButton -- props size(normal)', () => {
    const component = renderToString(<AtButton size='normal'>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props size(small)', () => {
    const component = renderToString(<AtButton size='small'>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props type(primary)', () => {
    const component = renderToString(<AtButton type='primary'>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props type(secondary)', () => {
    const component = renderToString(<AtButton type='secondary'>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props circle', () => {
    const component = renderToString(<AtButton circle>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props full', () => {
    const component = renderToString(<AtButton full>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props loading', () => {
    const component = renderToString(<AtButton loading>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })

  it('render AtButton -- props disabled', () => {
    const component = renderToString(<AtButton disabled>按钮</AtButton>)
    expect(component).toMatchSnapshot()
  })
})

describe('AtButton Event', () => {
  it('AtButton onClick', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(
      <AtButton onClick={onClick}>按钮</AtButton>
    )
    const componentDom = findDOMNode(component, 'at-button')

    Simulate.click(componentDom)
    expect(onClick).toBeCalled()
  })

  it('AtButton disabled, onClick not to be called', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(
      <AtButton disabled onClick={onClick}>
        按钮
      </AtButton>
    )
    const componentDom = findDOMNode(component, 'at-button')

    Simulate.click(componentDom)
    expect(onClick).not.toBeCalled()
  })
})
