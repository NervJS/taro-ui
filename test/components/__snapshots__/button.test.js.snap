// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtButton Snap render AtButton -- props circle 1`] = `"<div class=\\"at-button at-button--normal at-button--circle\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props disabled 1`] = `"<div class=\\"at-button at-button--normal at-button--disabled\\">undefinedundefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props full 1`] = `"<div class=\\"at-button at-button--normal at-button--full\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props loading 1`] = `"<div class=\\"at-button at-button--icon at-button--normal\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefined<div class=\\"at-button__icon\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props size(normal) 1`] = `"<div class=\\"at-button at-button--normal\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props size(small) 1`] = `"<div class=\\"at-button at-button--small\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props type(primary) 1`] = `"<div class=\\"at-button at-button--normal at-button--primary\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;

exports[`AtButton Snap render AtButton -- props type(secondary) 1`] = `"<div class=\\"at-button at-button--normal at-button--secondary\\"><button class=\\"at-button__wxbutton\\" lang=\\"en\\">undefined</button>undefinedundefinedundefined<div class=\\"at-button__text\\">按钮</div></div>"`;
